### PR TITLE
Feature nlp examples readme

### PR DESCRIPTION
**Pull request by InquestGeronimo**
NLP Transformer examples Readme update. I'm not sure how we expect to move forward with the pipelines in terms of discriminating between (token-classification and NER for example) so I guess I'll leave this up for the reviewers. This page is a bit tighter now. I removed the sparse zoo section towards the bottom as I felt it was too much content for this section.

### CODE REVIEW

File: examples/huggingface-transformers/README.md [hunk 1]
```diff
@@ -14,137 +14,117 @@ See the License for the specific language governing permissions and
...
-# Get DeepSparse question-answering pipeline
-
-qa_pipeline = pipeline(
-    task="question-answering",
-    model_path=onnx_filepath,
-    num_cores=num_cores,
+token_classification = pipeline(
+    task="token-classification",
+    model_path=model_path,
+    num_cores=None
```
Comment (mgoin) [inline examples/huggingface-transformers/README.md:36]:
```suggestion
```

File: examples/huggingface-transformers/README.md [hunk 2]
```diff
@@ -14,137 +14,117 @@ See the License for the specific language governing permissions and
...
 )
 
-# inference
-
-my_name = qa_pipeline(question="What's my name?", context="My name is Snorlax")
+inference = token_classification("I saw Snorlax in Texas!")
```
Comment (mgoin) [inline examples/huggingface-transformers/README.md:39]: what will this actually do/classify?

File: examples/huggingface-transformers/README.md [hunk 3]
```diff
@@ -14,137 +14,117 @@ See the License for the specific language governing permissions and
...
-    zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/pruned-aggressive_98 \
-    --batch-size 1
-```
 
+model_path = "zoo:nlp/text_classification/bert-base/pytorch/huggingface/sst2/base-none"
 
+text_classification = pipeline(
+    task="text-classification",
+    model_path=model_path,
+    num_cores=None
```
Comment (mgoin) [inline examples/huggingface-transformers/README.md:52]:
```suggestion
```

File: examples/huggingface-transformers/README.md [hunk 4]
```diff
@@ -14,137 +14,117 @@ See the License for the specific language governing permissions and
...
 
-### Server
+model_path="zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/pruned-aggressive_98"
 
-First, start up the host `server.py` with your model of choice, SparseZoo stubs are
-also supported.
+qa_pipeline = pipeline(
+    task="question-answering",
+    model_path=model_path,
+    num_cores=None,
```
Comment (mgoin) [inline examples/huggingface-transformers/README.md:68]:
```suggestion
```

File: examples/huggingface-transformers/README.md [hunk 5]
```diff
@@ -14,137 +14,117 @@ See the License for the specific language governing permissions and
...
+            alias: question_answering/dense
+        - task: question_answering
+            model_path: zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/pruned-aggressive_98
+            batch_size: 1
+            alias: question_answering/sparse_quantized
+
+After you finish building the `config.yaml` file, you can run the server with the config file path passed in the `--config_file` argument:
+```bash
+deepsparse.server --config_file config.yaml
+```
```
Comment (mgoin) [inline examples/huggingface-transformers/README.md:128]: To keep it simple maybe there is no need for multiple model inference here

File: examples/huggingface-transformers/README.md (hunk 2 above)
Comment (InquestGeronimo) [reply examples/huggingface-transformers/README.md:39]: it will print a list of dictionaries. Each dict will represent the classification of a token 
example of a single dict:
{'entity': 'LABEL_0', 'score': 0.9997255, 'index': 1, 'word': 'i', 'start': 0, 'end': 1}

One thing I want to point out, is that the `entity` variable doesn't have the label of the actual entity. This would be good if we can get these actual entities in the print out.

**Comment by dbarbuzzi** [comment]
For the final example using `deepsparse.server`, should it be mentioned somewhere that this is only available if installed via `pip install deepsparse[server]`?

File: examples/huggingface-transformers/README.md [hunk 6]
```diff
@@ -14,137 +14,93 @@ See the License for the specific language governing permissions and
...
-
-Available via API and hosted in the cloud, the [SparseZoo](http://sparsezoo.neuralmagic.com/) contains both 
-baseline and models sparsified to different degrees of inference performance vs. baseline loss recovery. Recipe-driven approaches built around sparsification algorithms allow you to take the models as given, transfer-learn from the models onto private datasets, or transfer the recipes to your architectures.
+For a more in-depth discussion on benchmarking, check out the [Benchmarking tutorial](https://github.com/neuralmagic/deepsparse/tree/main/src/deepsparse/benchmark_model)!
 
-Each model in the SparseZoo has a specific stub that identifies it. The stubs are made up of the following structure:
+### DeepSparse Server 🔌
 
-`DOMAIN/SUB_DOMAIN/ARCHITECTURE{-SUB_ARCHITECTURE}/FRAMEWORK/REPO/DATASET{-TRAINING_SCHEME}/SPARSE_NAME-SPARSE_CATEGORY-{SPARSE_TARGET}`
+The DeepSparse inference server allows you to serve ONNX models and pipelines in HTTP.
```
Comment (mgoin) [inline examples/huggingface-transformers/README.md:96]: Make sure to mention you need `pip install "deepsparse[server]"`, thanks @dbarbuzzi

**Review by mgoin — DISMISSED** [review]

File: examples/huggingface-transformers/README.md [hunk 7]
```diff
@@ -14,137 +14,97 @@ See the License for the specific language governing permissions and
...
-sparsified models from [SparseZoo](https://github.com/neuralmagic/sparsezoo) 
-or you can specify your own transformer [ONNX](https://onnx.ai/) file.
-
-## Installation
-The dependencies for these examples can be installed using `pip` and the supplied `requirements.txt` file:
-```bash
-pip3 install -r requirements.txt
-```
+## Transformers 🤗 Inference Pipelines
+This directory contains examples for serving, benchmarking, and running NLP models from the [Transformers](https://github.com/huggingface/transformers) repository using the DeepSparse Engine. These examples can load pre-trained, sparsified models from SparseZoo or you can specify your own transformer ONNX file. In addition, we also highlight how you can easily perform benchmarking and deploy transformers with the `deepsparse.server` via simple CLI commands.
```
Comment (rahul-tuli) [inline examples/huggingface-transformers/README.md:18]:
```suggestion
This directory contains examples for serving, benchmarking, and running NLP models from the [Transformers](https://github.com/huggingface/transformers) repository using the DeepSparse Engine. These examples can load pre-trained, sparsified models from SparseZoo, or you can specify your own transformer ONNX file. In addition, we also highlight how you can easily perform benchmarking and deploy transformers with the `deepsparse.server` via simple CLI commands.
```

**Review by rahul-tuli — APPROVED** [review]

**Review by mgoin — APPROVED** [review]

File: examples/huggingface-transformers/README.md [hunk 8]
```diff
@@ -14,137 +14,97 @@ See the License for the specific language governing permissions and
...
+    model_path=model_path,
+)
 
-Example command:
-```bash
-python server.py \
-    zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/pruned-aggressive_98
+inference = qa_pipeline(question="What's my name?", context="My name is Snorlax")
 ```
+💡**PRO-TIP**💡: The pipeline can also infer a default sparse model to run automatically:
```
Comment (jeanniefinks) [inline examples/huggingface-transformers/README.md:70]:
```suggestion
💡**PRO TIP**💡: The pipeline can also infer a default sparse model to run automatically:
```

File: examples/huggingface-transformers/README.md [hunk 9]
```diff
@@ -14,137 +14,97 @@ See the License for the specific language governing permissions and
...
-- `/info` to get information about the compiled model
-- `/predict` to send inputs to the model and receive a response.
-    The number of inputs should match the compiled model's batch size.
+qa_pipeline = pipeline("question-answering")
 
-For a full list of options, run `python server.py -h`.
+inference = qa_pipeline(question="What's my name?", context="My name is Snorlax")
+```
+**more tasks coming soon...😇**
+## Benchmarking & the DeepSparse Server
```
Comment (jeanniefinks) [inline examples/huggingface-transformers/README.md:80]:
```suggestion
## Benchmarking and the DeepSparse Server
```

File: examples/huggingface-transformers/README.md [hunk 10]
```diff
@@ -14,137 +14,97 @@ See the License for the specific language governing permissions and
...
 ```
 
-### SparseZoo Stubs
-[SparseZoo](http://sparsezoo.neuralmagic.com/) is a constantly-growing repository of sparsified (pruned and 
-pruned-quantized) models with matching sparsification recipes for neural networks. It simplifies and accelerates your time-to-value in building performant deep learning models with a collection of inference-optimized models and recipes to prototype from.
+For a more in-depth discussion on benchmarking, check out the [Benchmarking tutorial](https://github.com/neuralmagic/deepsparse/tree/main/src/deepsparse/benchmark_model)!
 
-Available via API and hosted in the cloud, the [SparseZoo](http://sparsezoo.neuralmagic.com/) contains both 
-baseline and models sparsified to different degrees of inference performance vs. baseline loss recovery. Recipe-driven approaches built around sparsification algorithms allow you to take the models as given, transfer-learn from the models onto private datasets, or transfer the recipes to your architectures.
+### DeepSparse Server 🔌
```
Comment (jeanniefinks) [inline examples/huggingface-transformers/README.md:94]: Did I see somewhere some of the icons coming before the headline? I think that's better as it has consistent placement when scanning a page (vs at the end of headlines with different lengths...)

**Review by jeanniefinks — CHANGES_REQUESTED** [review]
Thanks R!

**Review by mgoin — APPROVED** [review]
nice!

**Review by markurtz — DISMISSED** [review]

File: examples/huggingface-transformers/README.md (hunk 10 above)
Comment (InquestGeronimo) [reply examples/huggingface-transformers/README.md:94]: Ok i put the emojis at the beginning of headlines

File: examples/huggingface-transformers/README.md [hunk 11]
```diff
@@ -14,137 +14,97 @@ See the License for the specific language governing permissions and
...
 
-For a full list of options, run `python server.py -h`.
+inference = qa_pipeline(question="What's my name?", context="My name is Snorlax")
+```
+**more tasks coming soon...😇**
+## Benchmarking & the DeepSparse Server
 
-Currently, the server uses DeepSparse-HuggingFace pipeline integration 
-for end-to-end prediction.  
+### Benchmarking ONNX Models 📜
```
Comment (jeanniefinks) [inline examples/huggingface-transformers/README.md:82]:
```suggestion
###  📜 Benchmarking ONNX Models
```

File: examples/huggingface-transformers/README.md (hunk 10 above)
Comment (jeanniefinks) [inline examples/huggingface-transformers/README.md:94]:
```suggestion
###  🔌 DeepSparse Server
```

**Review by jeanniefinks — APPROVED** [review]
Lgtm

**Review by mgoin — APPROVED** [review]
